### DISCUSSION TITLE
Why Chatwoot requires advanced permissions on Facebook

### DISCUSSION THREAD

**Discussion by ghost** [chatwoot/chatwoot#7916] [Q&A] [answered]
I'm trying to be approved by Facebook review for Facebook Messenger and I need to know for what Chatwoot needs these advanced permissions:

- pages_messaging
- Business Asset User Profile Access
- pages_show_list
- pages_manage_metadata
- pages_read_engagement

I didn't find any clear text in Chatwoot docs about it.

I believe that with the reasons these permissions are used I can record a video and provide a text to Facebook.

Also, I read this [discussion response](https://github.com/orgs/chatroot/discussions/3397#discussioncomment-1911479) and again was unable to help me with this problem.

**Comment by slamer59** [comment]
I face the same issue.
Did you solve it ? 

The scopes are defined here : 
https://github.com/chatwoot/chatwoot/blob/682a2aea1cdc89fd1432490d0b6e55b103322a4b/app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Facebook.vue#L235

This is my 2 cents : 
- pages_show_list : Chatwoot needs to know which facebook page can be accessed by the user. Channel -> user connection -> FB page
- pages_messaging : interact between operator and end user. 

- Business Asset User Profile Access : Is the operator linked to the Meta Business ?

- pages_manage_metadata : required for pages_messaging
- pages_read_engagement : chatwoot needs to have access to page content.

**Comment by EmanoelDionisio** [comment]
They are currently requiring a visual demonstration and I can't justify this visually to them regarding this permission, how could we do this in a practical way for the pages_read_engagement functionality?

**Reply by Achrafchrouda** [reply]
yes can someone tell us , how we can excatly show this permissions using to facebook ?

**Comment by Achrafchrouda** [comment]
any update here ?

**Comment by Witroch4** [comment]
help

**Comment by Witroch4** [comment]
Any update?

**Comment by badereddineqodia** [comment]
any update ?

**Comment by lzkill** [comment]
Any updates? This app review process from Meta is a complete blocker for everyone trying to use Chatwoot. There must be some templates available to help pass through this with less hassle.

What exactly must we provide to get these permissions approved?

instagram_manage_messages
instagram_basic
pages_show_list
pages_manage_metadata
pages_messaging
business_management
pages_read_engagement

**Comment by ZenDevMaster** [comment]
If your business is verified on Facebook, it's possible to use Chatwoot - in Facebook App Live Mode - on your own facebook/whatsapp/insta accounts. 

You don't need to register as a Tech Provider or go through full advanced verification on all these permissions unless you will have 3rd parties clients link their FB/IG/WA accounts to your Chatwoot install. 

I believe the only limit will be "John Doe" user names in Facebook Messenger if you've not obtained "pages_messaging" advanced permission and "Business Asset User Profile Access"

**Comment by muhsin-k** [comment, marked as answer]
Here’s a clear summary of the Messenger-related permissions required for Chatwoot:

1. **pages_show_list**
    
    **Purpose:** Fetches the list of Facebook Pages the user manages.
    
    **Why it's needed:** Allows users to select which Page they want to connect to Chatwoot for Messenger support.
    
2. **pages_messaging**
    
    **Purpose:** Enables your app to send and receive messages on behalf of a connected Facebook Page via Messenger.
    
    **Why it's needed:** Provides the core functionality that allows agents to respond to customer messages.
    
3. **pages_manage_metadata**
    
    **Purpose:** Allows your app to subscribe to and receive Messenger webhook events for a Facebook Page (e.g., new messages, delivery receipts).
    
    **Why it's needed:** Ensures real-time message delivery and updates between Messenger and the Chatwoot Inbox.
    
4. **business_management**
    
    **Purpose:** Grants your app access to Business Manager assets and verifies Page ownership.
    
    **Why it's needed:** Supports secure onboarding by confirming that the Page is properly managed by the user's Business account before connecting it to the Inbox.
    
5. **Business Asset User Profile Access**
    
    **Purpose:** Retrieves the name, profile picture, and IDs of users who message the connected Page.
    
    **Why it's needed:** Displays customer identity inside Chatwoot so agents know who they’re talking to, improving response personalization.
    

These permissions cover all essential workflows required to manage Messenger conversations.

**Note:** We currently do not require the `pages_read_engagement` permission unless you plan to read Page posts or pull public insights. This permission will be removed from our codebase soon.

cc @sojan-official

**Reply by pranavrajs** [reply]
@muhsin-k Let add it to the developer doc as well.

**Reply by fabwnb** [reply]
Meta is asking for a screencast. Have you got the screencasts to share as well? It would be very helpful and thank you in advance to the summary.
